### ISSUE TITLE
Recent activity API ERROR

### ISSUE THREAD

**Issue by ghost** [mx-space/core#1671]
### Describe the bug

访问 recent API 报错 `"Cannot use 'in' operator to search for 'nid' in undefined"`
![image](https://github.com/mx-space/core/assets/168252381/130c196e-d8e4-4062-9275-12b0fd2d12cc)

### Reproduction

Call `recent` API

### System Info

```Shell
Environment:  
 System:
    OS: Linux 6.1 Alpine Linux
    CPU: (2) x64 Intel(R) Xeon(R) Platinum 8255C CPU @ 2.50GHz
    Memory: 848.11 MB / 1.80 GB
    Container: Yes
    Shell: 1.36.1 - /bin/ash
  Binaries:
    Node: 20.12.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.5.0 - /usr/local/bin/npm
API Endpoint: https://blog.ayo.moe/api/v2
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by Innei** [comment]
无法复现，提供具体信息

**Comment by ghost** [comment]
> 无法复现，提供具体信息

![image](https://github.com/mx-space/core/assets/168252381/29fd98c2-29c9-4f0e-b858-71e2fe0b33e7)
 ```
ERROR   [Catch]  Cannot use 'in' operator to search for 'nid' in undefined                                                                                         1:21:41 PM

  at ActivityController.getRecentActivities (src/modules/activity/activity.controller.ts:183:16)
  at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by Innei** [comment]
你本地调试的数据方便脱一份给我吗

**Comment by ghost** [comment]
> 你本地调试的数据方便脱一份给我吗

我直接拿生产环境的数据复现的 可以发送到 `i@innei.in` 吗

**Comment by Innei** [comment]
收到了

**Comment by Innei** [comment]
存在脏数据，ref 的数据被删除了 ，我这里兜个底

**Comment by Innei** [comment]
Fixed in 911640b32b26711670b68fd0b87bd0884c53df2a

**Comment by Innei** [comment]
Fixed in 911640b32b26711670b68fd0b87bd0884c53df2a
